games/TicTacToe/project/tictactoe-contract/src/main.sw

```diff
@@ -0,0 +1,113 @@
+contract;
+
+dep interface;
+dep events;
+dep errors;
+dep utils;
+dep data_structures;
+
+use interface::Game;
+use events::{GameDrawnEvent, GameWonEvent, NewGameEvent};
+use errors::{GameStateError, PlayerError, PositionError};
+use utils::{draw, win_check};
+use data_structures::State;
+use core::ops::Eq;
+use std::{auth::msg_sender, logging::log};
+
+// This is needed for comparing the position when the cell is not empty.
+// We only need to check if there is an Identity in the cell but we don't care of its value.
+impl<T> Eq for Option<T> {
+    fn eq(self, other: Self) -> bool {
+        match (self, other) {
+            (Option::None, Option::None) => true,
+            (Option::Some(T), Option::Some(T)) => true,
+            _ => false,
+        }
+    }
+}
+
+storage {
+    /// The first player of the game.
+    player_one: Option<Identity> = Option::None,
+    /// The second player of the game.
+    player_two: Option<Identity> = Option::None,
+    /// Keeps track of each player move.
+    board: StorageMap<u64, Option<Identity>> = StorageMap {},
+    /// The current player turn.
+    player_turn: Option<Identity> = Option::None,
+    /// Keeps track of the move counter for various checks (win, draw, etc.).
+    move_counter: u64 = 0,
+    /// Keeps track of the game, its value is either Ended or Playing.
+    state: State = State::Ended,
```

Comment:
another consideration: perhaps we want to support more than one game concurrently, in which case we could store a `Game` struct which contains stuff like the players, the status, etc...

Reply:
Yeap; however, for a first addition a simple game is fine. We can make improvements with further PRs